control-userlayer/src/main/java/org/oskari/control/userlayer/CreateUserLayerHandler.java

```diff
@@ -346,12 +377,8 @@ private static String getFileExt(String name) {
 
     private SimpleFeatureCollection parse(FeatureCollectionParser parser, File file,
             CoordinateReferenceSystem sourceCRS,
-            CoordinateReferenceSystem targetCRS) throws ActionException {
-        try {
-            return parser.parse(file, sourceCRS, targetCRS);
-        } catch (ServiceException e) {
-            throw new ActionException("Failed to parse Feature Collection", e);
-        }
+            CoordinateReferenceSystem targetCRS) throws ServiceException {
+        return parser.parse(file, sourceCRS, targetCRS);
```

Comment:
This function isn't needed anymore since it only transformed the ServiceException into a ActionException